javaslang/generator/Generator.scala

```diff
@@ -1265,46 +1265,40 @@ def generateMainClasses(): Unit = {
 
             ${(i == 0).gen(xs"""
               @Override
-              public <T> Tuple1<T> prepend(T value) {
+              public <T> Tuple1<T> append(T value) {
                  return new Tuple1<>(value);
               }
 
               @Override
-              public <T> Tuple1<T> append(T value) {
+              public <T> Tuple1<T> prepend(T value) {
                  return new Tuple1<>(value);
               }
             """)}
 
             ${(i > 0 && i < N).gen(xs"""
               @Override
-              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
-                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
-              }
-
-              @Override
               public <T> Tuple${i + 1}<${(1 to i).gen(j => s"T$j")(", ")}, T> append(T value) {
                   return Tuple.of(${(1 to i).gen(j => s"_$j")(", ")}, value);
               }
-            """)}
 
-            ${(i == N).gen(xs"""
               @Override
-              public <T> Tuple prepend(T value) {
-                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
+              public <T> Tuple${i + 1}<T, ${(1 to i).gen(j => s"T$j")(", ")}> prepend(T value) {
+                  return Tuple.of(value, ${(1 to i).gen(j => s"_$j")(", ")});
               }
+            """)}
 
+            ${(i == N).gen(xs"""
               @Override
               public <T> Tuple append(T value) {
                   throw new UnsupportedOperationException("Append to Tuple${N}");
               }
-            """)}
 
-            ${(i > 0).gen(xs"""
-              public static $generics $className<${(1 to i).gen(j => s"Seq<? extends T$j>")(", ")}> sequence(Iterable<$className<${(1 to i).gen(j => s"? extends T$j")(", ")}>> tuples) {
-                Objects.requireNonNull(tuples, "tuples is null");
-                return new $className<>(${(1 to i).gen(j => s"${im.getType("javaslang.collection.Iterator")}.ofAll(tuples).map($className::_$j).toList()")(", ")});
+              @Override
+              public <T> Tuple prepend(T value) {
+                  throw new UnsupportedOperationException("Prepend to Tuple${N}");
```

Comment:
It looks like append() and prepend() are Tuple builders. In your example the elements are already known. Changing the example to the following would more concise but also inefficient regarding creation of Tuple instances:

```java
Tuple result = Tuple.empty();
for (int i = 0; i < list.size(); i++) {
    result = result.append(list.get(i));
}
```

Because append() and prepend() make most sense in a dynamic context (i.e. size of the List is unknown), the result type is just `Tuple`. In my opinion this kind of code is a corner case. It is comparable / has the same characteristics like reflection, which is a code smell and not type safe.

I'm not convinced that these methods are really useful, they do not encourage a good programming practice.